extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -524,6 +531,51 @@ export class AmpConsent extends AMP.BaseElement {
     });
   }
 
+  /**
+   * Sync consent string and consent state value to local storage.
+   */
+  syncConsentStringAndStateValue_() {
+    this.getConsentRemote_().then(response => {
+      // Decision from promptUI takes precedence over response
+      if (response && !this.consentStateChangedViaPromptUI_) {
+        this.consentStateManager_
+          .getLastConsentInstanceInfo()
+          .then(storedInfo => {
+            const consentRequired = response['consentRequired'];
+            const consentState = convertValueToState(
+              response['consentStateValue']
+            );
+            // Local storage empty, consent required, and state is not unknown,
+            // then use the state given.
+            if (
+              !!consentRequired &&
+              consentState !== CONSENT_ITEM_STATE.UNKNOWN
+            ) {
+              this.updateCacheIfNotNull_(storedInfo, response);
+            }
+          });
+      }
+    });
+  }
+
+  /**
+   * Sync cache is the value is not null. Defaults to storedValue.
+   * @param {./consent-info.ConsentInfoDef} storedInfo
+   * @param {?JsonObject} response
+   */
+  updateCacheIfNotNull_(storedInfo, response) {
```

Comment:
`storedInfo` is only used in this method, can be fetched here. 
I would update the method signature to:

```
updateCacheIfNotNull_(consentStateValue, consentString)
```